### PR TITLE
CD-37836 Add support for Rails 4.2

### PR DESCRIPTION
**Pull request by johnny-lai**
Fix is based on:
* https://github.com/clemens/delocalize/issues/74
* https://gist.github.com/daniel-rikowski/fd09dc2cc82ce28e7986

- [x] @torumori 
- [x] @edk 
please review

### CODE REVIEW

**Comment by johnny-lai** [comment]
Fix is basically the `delocalize/rails_ext/active_record_rails4.rb` + `https://gist.github.com/daniel-rikowski/fd09dc2cc82ce28e7986`

I also had to remove https://github.com/coupa/delocalize/blob/0-4-coupa/lib/delocalize/rails_ext/active_record_rails4.rb#L66 because that variable was not in Rails 4.2. I checked the code and the writers looked like they just call `write_attribute`.

**Comment by torumori** [comment]
I've been wondering if we should upgrade our forked version to 0.4.0, but looks like we still need this monkey patch even if we do it.
And changing to 1.0-beta seems to be more work on our side.
So, I still feel we might want to upgrade to 0.4.0 (we need to apply some of our own patches) in our Rails 4.2 upgrade work, but I won't insist :smile_cat:

**Comment by johnny-lai** [comment]
@torumori fyi, it looks like we are using the 0.4.0. the 0-4-coupa branch has the last commit made to 0.4 276351efa132aaeab733504d932362f698608f90

**Comment by torumori** [comment]
@johnny-lai Oh, sorry, I was looking at R12 branch which uses master branch. Looks like it was already upgraded during Rails 4 work in R13. Thanks for the confirmation.

File: lib/delocalize/rails_ext/active_record_rails42.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        end
+      EOV
+      generated_attribute_methods.module_eval(method_body, __FILE__, line)
+    else
+      super
+    end
+  end
+end
+
+module ActiveRecord
```
Comment (edk) [inline lib/delocalize/rails_ext/active_record_rails42.rb]: this section came from the referenced gist.  Where did the code above come from?

Comment (edk) [inline lib/delocalize/rails_ext/active_record_rails42.rb]: overall, it looks ok.  Do you know if we've got any unit testing to verify the delocalize stuff?  I'm guessing at the very least the full stack integration tests probably do things in different locales that should cover this here.

**Comment by edk** [comment]
:+1:

**Comment by johnny-lai** [comment]
So it does come with tests. The tests are failing atm, which is why I havn't merged yet

**Pull request merged**